Xamarin/AndroidClient/MainActivity.cs

```diff
@@ -18,74 +28,160 @@ protected override void OnCreate(Bundle savedInstanceState)
-            var connectButton = FindViewById<Button>(Resource.Id.bConnect);
-            var listView = FindViewById<ListView>(Resource.Id.lvMessages);
-            var sendButton = FindViewById<Button>(Resource.Id.bSend);
-            var messageText = FindViewById<EditText>(Resource.Id.etMessageText);
-            var connected = false;
-
-            var hubConnection = new HubConnectionBuilder()
-                    .WithUrl("<YOUR SERVER URL HERE>")
-                    .Build();
-
-            //Setting up window to enter name
-            AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
-            alertDialogBuilder.SetTitle("Enter Your Name");
-
-            var userName = "";
-            var nameInput = new EditText(this);
-            nameInput.InputType = Android.Text.InputTypes.TextVariationPersonName;
-            alertDialogBuilder.SetView(nameInput);
-
-            alertDialogBuilder.SetPositiveButton("Ok", (_, __) =>
-            {
-                userName = nameInput.Text;
-            });
+            _connectButton = FindViewById<Button>(Resource.Id.bConnect);
+            _listView = FindViewById<ListView>(Resource.Id.lvMessages);
+            _sendButton = FindViewById<Button>(Resource.Id.bSend);
+            _messageText = FindViewById<EditText>(Resource.Id.etMessageText);
+            _serverUrlText = FindViewById<EditText>(Resource.Id.etServerUrl);
 
-            alertDialogBuilder.SetNegativeButton("Cancel", (_, __) => { });
+            _sendButton.Click += SendButton_Click;
+            _connectButton.Click += ConnectButton_Click;
 
-            alertDialogBuilder.Show();
+            _messages = new ArrayAdapter(this, Android.Resource.Layout.SimpleListItem1, new List<string>());
+            _listView.Adapter = _messages;
 
-            var messages = new List<string>();
-            var arrayAdapter = new ArrayAdapter(this, Android.Resource.Layout.SimpleListItem1, messages);
-            listView.Adapter = arrayAdapter;
+            ////Setting up window to enter name
+            //AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+            //alertDialogBuilder.SetTitle("Enter Your Name");
 
-            connectButton.Click += async (sender, e) =>
-            {
-                await hubConnection.StartAsync();
-                connectButton.Enabled = false;
-                connected = true;
-                Toast.MakeText(this, "Connected", ToastLength.Short).Show();
-                await hubConnection.InvokeAsync("Send", userName, "is now connected");
-            };
-
-            hubConnection.On<string, string>("broadcastMessage", (name, message) =>
-            {
-                var newMessage = $"{name}: {message}";
-                RunOnUiThread(() => {
-                    arrayAdapter.Add(newMessage);
-                    arrayAdapter.NotifyDataSetChanged();
-                });
-            });
+            //var userName = "";
```

Comment:
Remove?

Reply:
Yeah, I actually had a bigger plan to have the SignalRSamples support the user name (since connection IDs are gross), but that didn't make it. I'll remove this goop for now :)